### PR TITLE
feat(grid): add display density option to the grid

### LINKED ISSUE #1320: igx-Grid visual density: Compact mode
### Description  
The igx-grid needs to have an easy way to be visually optimised for data-heavy enterprise (financial, medical, etc.) scenarios. 

The change should apply to all parts of the grid: 
 - Cells 
 - Header Cells
 - Summaries
 - Paging

### Suggested sizes/paddings at `compact` version
- left/right paddings for all cells: `12px`  *(down from 24px)*
- default row height: `32px`  *(down from 48px)*
- header height: `32px`
- summaries vertical padding: `0`  *(down from 5px)*

### PR DESCRIPTION
**Pull request by simeonoff**
Closes #1320

### CODE REVIEW

File: src/grid/grid.component.ts [hunk 1]
```diff
@@ -482,6 +511,7 @@ export class IgxGridComponent implements OnInit, OnDestroy, AfterContentInit, Af
         this.calcWidth = this._width && this._width.indexOf("%") === -1 ? parseInt(this._width, 10) : 0;
         this.calcHeight = 0;
         this.calcRowCheckboxWidth = 0;
+        this.rowHeight = this.rowHeight ? this.rowHeight : this.defaultRowHeight;
```
Comment (kdinev) [inline src/grid/grid.component.ts]: In this case the `rowHeight` overrides the `displayDensity` option. Maybe we should actually make the `rowHeight` private and control the height only through the `displayDensity` internally? This way we would give less control, but be sure that the grid looks optimal.

Comment (simeonoff) [reply src/grid/grid.component.ts]: I believe rowHeight should remain public, regardless of the displayDensity. When you set rowHeight explicitly as a user you change the height of the row and knowingly stray from the recommended default value we set. The displayDensity controls the cell padding as well.

**Comment by ddincheva** [comment]
there is two failing test because of issue [1305](https://github.com/IgniteUI/igniteui-angular/issues/1305)

**Comment by zdrawku** [comment]
we are currently working on the falling tests

**Pull request merged**